L1Trigger/TrackTrigger/interface/HitPatternHelper.h

```diff
@@ -0,0 +1,169 @@
+// This is a helper function that can be used to decode hitpattern, which is a 7-bit integer produced by KF.
+//
+// There are three classes declared in HitPatternHelper (HPH) namesapce:
+// 1)SensorModule: This is used to store important information about the sensor modules. For example r,z coordinates.
+// 2)Setup: This is used to produce a collection of <SensorModule> needed by HPH.
+// 3)HitPatternHelper: This is used to decode hitpattern with the help of the information from sensor modules and layermap.
+//
+// Two predictions on which layers particles will hit are made using different information:
+// i)Loop over sensor modules and make predictions based on spatial coordinates of tracks. This prediction is considered more accurate.
+// ii)Make predictions based on a hard-coded layermap. This prediction is considered less accurate and is used by Old KF to encode hitpattern.
+//
+//
+//  Created by J.Li on 1/23/21.
+//
+
+#ifndef L1Trigger_TrackTrigger_interface_HitPatternHelper_h
+#define L1Trigger_TrackTrigger_interface_HitPatternHelper_h
+
+#include "FWCore/Framework/interface/data_default_record_trait.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "Geometry/CommonTopologies/interface/PixelGeomDetUnit.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "L1Trigger/TrackTrigger/interface/HitPatternHelperRcd.h"
+
+#include <bitset>
+#include <iostream>
+#include <vector>
+
+using namespace std;
+using namespace edm;
+
+namespace hph {
+
+  class SensorModule {
```

Comment:
There is already a class SensorModule.h (written by Thomas Schuh) in L1Trigger/TrackerDTC/ (which he moves to L1Trigger/TrackTrigger/ in https://github.com/cms-L1TK/cmssw/pull/88 ). Can we not use this one instead of creating a new class? Thomas can explain how it is initialized for all modules in Tracker.

Reply:
I intend to make this utility tool a standalone package with few dependencies. That is why I create this lighter version of the tracker module class. A lot of the information stored at the two options you referenced is not really needed by this package so I wonder if it is really necessary to use these alternative implementations. 
I can maybe switch to Thomas's module function If you have a strong preference for this. In that case, we need to add "maybe layer" data member to his class.

Reply:
Almost all the functions in HitPatternHelper::SensorModule are already defined in https://github.com/cms-L1TK/cmssw/blob/L1TK-dev-12_0_0_pre4_HPH/L1Trigger/TrackTrigger/interface/SensorModule.h . The latter function is filled and made available by the ESProducer L1Trigger/TrackTrigger/python/ProducerSetup_cff.py . We should avoid having two almost identical classes, so talk to @tschuh about using the existing code.

Reply:
HitPatternHelper::SensorModule has been removed